### PR TITLE
Update BUILD.md

### PR DESCRIPTION
**Pull request by geoffthemedio**
-link to SDK13
-tweak Visual Studio URL
-update versions

### CODE REVIEW

**Comment by geoffthemedio** [comment]
@o01eg does this look accurate to you?

**Comment by o01eg** [comment]
@geoffthemedio Yes, looks fine. Still it quite sad memory requirements now twice of previous.

**Comment by geoffthemedio** [comment]
> memory requirements now twice of previous.

Probably not actually needed unless running parallel builds...

File: BUILD.md [hunk 1]
```diff
@@ -51,21 +51,21 @@ FreeOrion depends on the following libraries or APIs to run:
 Obtaining FreeOrion Source Code and Software Dependencies
 ---------------------------------------------------------
 
-For Windows and Mac OS X a [Software Development Kit] is provided as download to
-compile FreeOrion from source. It contains the preconfigured and -compiled build
-and runtime dependencies for the Visual Studio v141 toolchain on Windows and
+For Windows and Mac OS X, a [Software Development Kit] is provided as download to
+assist with compiling FreeOrion from source. The SDK contains the preconfigured and
+-compiled build and runtime dependencies for Visual Studio on Windows and the
 Mac OS X 10.9 SDK with Xcode 6.4 or later on Mac OS X.
```
Comment (Vezzra) [inline BUILD.md]:
```suggestion
macOS 10.12 SDK with Xcode 10.1 or later on macOS.
```

File: BUILD.md [hunk 2]
```diff
@@ -3,19 +3,19 @@ Building FreeOrion
...
 These notes are applicable for building the master development Git branch
 of FreeOrion. For building numbered FreeOrion releases, consult the
-corresponding release branch, eg. [FreeOrion v0.4.10 BUILD.md]
+corresponding release branch, eg. [FreeOrion v0.5 BUILD.md]
 
 Hardware and OS Requirements
 ----------------------------
 
-FreeOrion should compile on Windows 7 (or later), Mac OS X 10.10 (or later) and
+FreeOrion should compile on Windows 8.1 (or later), Mac OS X 10.10 (or later) and
```
Comment (Vezzra) [inline BUILD.md]:
```suggestion
FreeOrion should compile on Windows 8.1 (or later), macOS 10.12 (or later) and
```

**Review by Vezzra — CHANGES_REQUESTED** [review]
In addition to the suggested changes, all "Mac OSX" / "Mac OS X" etc. occurences should be replaced by "macOS".

**Review by Vezzra — APPROVED** [review]
LGTM.

As this is marked for cherry picking, shall I merge this before I produce RC2 on Tuesday (if it hasn't been merged by then)?

**Comment by geoffthemedio** [comment]
yes... updated instructions are relevant for v0.5

**Pull request merged**